modules/videoio/src/cap_msmf.cpp

```diff
@@ -853,31 +857,28 @@ bool CvCapture_MSMF::configureOutput(UINT32 width, UINT32 height, double prefFra
             MediaType MT(pType.Get());
             if (MT.MF_MT_MAJOR_TYPE == MFMediaType_Video)
             {
-                if (dwStreamFallback < 0 ||
-                    ((MT.width * MT.height) > (MTFallback.width * MTFallback.height)) ||
-                    (((MT.width * MT.height) == (MTFallback.width * MTFallback.height)) && getFramerate(MT) > getFramerate(MTFallback) && (prefFramerate == 0 || getFramerate(MT) <= prefFramerate)))
-                {
-                    dwStreamFallback = (int)dwStreamTest;
-                    MTFallback = MT;
-                }
-                if (MT.width == width && MT.height == height)
+                if (dwStreamBest < 0 ||
+                    (
+                        ((width > MT.width ? width - MT.width : width > 0 ? MT.width - width : 0) + (height > MT.height ? height - MT.height : height > 0 ? MT.height - height : 0)) <
```

Comment:
the conditions look too complex. Can we simplify them?
e.g.

<pre>
int best_diff = INT_MAX;

for( ... ) {
...
int temp_width = width > 0 ? width : MT.width;
int temp_height = height > 0 ? height : MT.height;
int diff = abs(temp_width - MT.width) + abs(temp_height - MT.heighT);
if( diff < best_diff ) { best_width = MT.width; best_height = MT.height; best_diff = diff; }
}
</pre>

Reply:
I've refactored the code. Now it should be more readable